pkg/policy/api/rules.go

```diff
@@ -14,9 +14,30 @@
 
 package api
 
+import (
+	"github.com/cilium/cilium/pkg/labels"
+)
+
 // Rules is a collection of api.Rule.
 //
 // All rules must be evaluated in order to come to a conclusion. While
 // it is sufficient to have a single fromEndpoints rule match, none of
 // the fromRequires may be violated at the same time.
 type Rules []*Rule
+
+// GetSetOfLabels returns the the set of labels defined in all rules of `rs`.
+func (rs Rules) GetSetOfLabels() []labels.LabelArray {
```

Comment:
Please add a warning to be extremely careful with this function. This has a potential runtime worst case of O(n^4)

Reply:
Same here. Do you still want to have the warning here as I've changed the Equals?